src/Storages/VirtualColumnUtils.h

```diff
@@ -47,7 +50,7 @@ auto extractSingleValueFromBlock(const Block & block, const String & name)
 }
 
 NameSet getVirtualNamesForFileLikeStorage();
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns);
+VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns, std::string path = "", FormatSettings settings = FormatSettings());
```

Comment:
It's not good to use default format settings. If we don't have format settings we should create it from context.

So, it will be something like:
```
VirtualColumnsDescription getVirtualsForFileLikeStorage(
    const ColumnsDescription & storage_columns,
    const ContextPtr & context,
    std::string sample_path = "",
    std::optional<FormatSettings> format_settings_ = std::nullopt);
``` 

And then we can do:
```
    auto format_settings = format_settings_ ? *format_settings_ : getFormatSettings(context);
```

Also, as we will propogate the context, we can move checking the setting `use_hive_partitioning` inside `getVirtualsForFileLikeStorage`